packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -29,25 +29,50 @@ export function FileUploader({
     onDragStart,
     onDrop,
     setShowPreviewer,
-    setTargetFiles,
+    addTargetFiles,
     showPreviewer,
+    setFiles,
   } = useFileUploader();
+  const [allFileNames, setAllFileNames] = useState<string[]>([]);
 
   useEffect(() => {
     setShowPreviewer(isPreviewerVisible);
   }, [setShowPreviewer, isPreviewerVisible]);
 
+  useEffect(() => {
+    setAllFileNames(files.map((file) => file.name));
+  }, [files]);
+
```

Comment:
Not sure if there is a better way of doing this, but I added in two useEffects because combining them into one was causing the setShowPreviewer to be called whenever a file changed. Since it's setting the value from the prop, it would make the previewer disappear. So I split them.